### DISCUSSION TITLE
Network masks/prefixes

### DISCUSSION THREAD

**Discussion by Azmodeszer** [bb-Ricardo/netbox-sync#352] [Q&A] [answered]
What CIDR notation does the sync follow exactly? Netbox itself uses ex. 192.168.1.2/24 for a single specific address in a /24 network (as opposed to 192.168.1.2/32), but I've recently started to adjust some of our IP objects still following the other method and I noticed that the plugin is trying to "fix" some of those back to /32 masks. Shouldn't it get the prefix information from vCenter and apply it correctly? It appears to write a lot of addresses with a /32 mask, which is not correct for all but a select few special cases.

Or is the problem with how the subnetting is mapped in vCenter and the plugin is simply relaying what it is given?

**Comment by bb-Ricardo** [comment]
Hi,

It usually uses whatever was returned from vCenter. Do you have any log examples where these IP addresses are picked up?

**Reply by Azmodeszer** [reply]
Here's a few from the log, from both private and public address spaces:
```
INFO: Ip address '<ip>/24' attribute 'address' changed from '<ip>/21' to '<ip>/24'
INFO: Ip address '<ip>/32' attribute 'address' changed from '<ip>/21' to '<ip>/32'
INFO: Ip address '<ip>/32' attribute 'address' changed from '<ip>/21' to '<ip>/32'
INFO: Ip address '<ip>/32' attribute 'address' changed from '<ip>/21' to '<ip>/32'
[..]
INFO: Ip address '<ip>/24' attribute 'address' changed from '<ip>/28' to '<ip>/24'
INFO: Ip address '<ip>/32' attribute 'address' changed from '<ip>/27' to '<ip>/32'
[..]
INFO: Ip address '<ip>/28' attribute 'address' changed from '<ip>/30' to '<ip>/28'
```

The masks on the left are all edited by me in accordance with what it should be. Notice that it doesn't only change things to /32, but also /24 or /28, so I suppose I'll need to have a look at the network structure in vCenter.

**Reply by bb-Ricardo** [reply]
Maybe try to switch to DEBUG2 mode. And try to see what `ip a` returns as interface addresses inside your linux VMs.

**Reply by Azmodeszer** [reply, marked as answer]
Yeah, just checked the IP and mask directly on one of the affected VMs that I have access to and it does indeed show a /32 mask. So that's just a discrepancy between the info entered in Netbox and the state of our network. Not a Netbox/plugin issue, then.

Thanks.
